coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/functions/CoalesceStructUtilityTest.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2018-2021 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.hive.hive2rel.functions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.linkedin.coral.common.ToRelConverterTestUtils;
+import com.linkedin.coral.hive.hive2rel.TestUtils;
+
+import static com.linkedin.coral.hive.hive2rel.functions.CoalesceStructUtility.coalesce;
+import static com.linkedin.coral.hive.hive2rel.functions.CoalesceStructUtility.isTrinoStructPattern;
+
+
+public class CoalesceStructUtilityTest {
+  private RelDataTypeFactory typeFactory;
+  private RelDataType trinoStruct;
+  private RelDataType extractUnionStruct;
+  private RelDataType nonTrinoStruct;
+
+  @BeforeClass
+  public void setup() throws Exception {
+    ToRelConverterTestUtils.setup(TestUtils.loadResourceHiveConf());
+    typeFactory = ToRelConverterTestUtils.createRelBuilder().getTypeFactory();
+
+    List<String> names = ImmutableList.of("tag", "field0", "field1");
+    List<RelDataType> types = ImmutableList.of(typeFactory.createSqlType(SqlTypeName.INTEGER),
```

Comment:
I think it should be `SqlTypeName.TINYINT`, although it wouldn't affect the test.